packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -420,6 +436,7 @@ const queryStatePreSelector = (
   return {
     ...currentState,
     data,
+    currentData: currentState.data,
```

Comment:
Correct me if I am wrong, but wouldn't this be basically the equivalent of doing:
```typescript
const currentData = isFetching ? data : undefined;
```
in our own code? I would think you'd need to examine the `args` associated with `lastResult` to see if they are shallowly equal to the args of `currentState`, something like:

```
currentData = currentState.data ?? (lastResult?.arg === currentState.arg ? lastResult?.data : undefined)
```
if the comment 
> this property will always contain the received data from the query, for the *current query arguments*

is to be true. (i.e., distinguishing cases of polling from cases of arg changes).

But, correct me if I am wrong here. I am not too familiar with the internal workings of this codebase.

Reply:
`currentState` is always for the currently selected endpoint and `lastResult` is the last result the hook had cached. So `lastResult` can also be from another `arg`, but `currentState` never can.

Reply:
@phryneas Right... but my question is: this solution doesn't seem to distinguish between polling and arg changes. 
The behavior I'd expect for `currentData` is that if the args haven't changed (e.g., because of polling), then it would use the last result for those args (which might be cached in lastResult in that case).

Reply:
Data once present for the current args will not disappear from the cache while polling, and this essentially gives you data for the current args directly from the cache. 
Once data is set in the cache, it will always be held, even while loading or an error occurs later. 
See the slice definition here: https://github.com/reduxjs/redux-toolkit/blob/261410a91eb095a7611550a464117d3993d14b44/packages/toolkit/src/query/core/buildSlice.ts#L121-L169